### PR TITLE
don't pass `--memory-swappinness 0` if parameter is omitted

### PR DESCRIPTION
**Pull request by bjaglin**
Fixes compatibility with engine<1.8.0 and wrongly-forced option on engine>=1.8.0.
See https://github.com/michaelsauter/crane/pull/244.

### CODE REVIEW

File: crane/container.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ type RunParameters struct {
 	RawMemory            string      `json:"memory" yaml:"memory"`
 	RawMemoryReservation string      `json:"memory-reservation" yaml:"memory-reservation"`
 	RawMemorySwap        string      `json:"memory-swap" yaml:"memory-swap"`
-	MemorySwappiness     int         `json:"memory-swappiness" yaml:"memory-swappiness"`
+	MemorySwappiness     OptInt      `json:"memory-swappiness" yaml:"memory-swappiness"`
```
Comment (bjaglin) [inline crane/container.go]: we are loosing type checks with that unfortunately...

**Comment by michaelsauter** [comment]
Thanks a lot for this! I didn't test on earlier Docker versions either. The whole test setup is not the best :(

Anyway. Maybe we should just raise the minimum Docker version more frequently, otherwise this will become unmanageable.

**Comment by michaelsauter** [comment]
I'll release this as 2.3.1 tonight.

**Comment by bjaglin** [comment]
OK, thanks, ping me if I can help for the release.

**Comment by michaelsauter** [comment]
Thanks - released it now: https://github.com/michaelsauter/crane/releases/tag/v2.3.1

**Pull request merged**